others/sub_sequences.cpp

```diff
@@ -0,0 +1,23 @@
+/* Copyright 2019 by Contributors */
+#include <cmath>
+#include <iostream>
+ 
+void sub_sequences(char *in, int i, char *out, int j) {
```

Comment:
please add description of the algorithm, iterative, recursive, etc?

Reply:
since `in` variable is not changing, per C++ guidelines, it should be passed in as a constant pointer. It improves the code efficiency